src/mantarray_desktop_app/mc_simulator.py

```diff
@@ -170,12 +154,8 @@ def _handle_comm_from_pc(self) -> None:
             return
 
         # validate checksum before handling the communication
-        expected_checksum = crc32(comm_from_pc[:-SERIAL_COMM_CHECKSUM_LENGTH_BYTES])
-        actual_checksum = int.from_bytes(
-            comm_from_pc[-SERIAL_COMM_CHECKSUM_LENGTH_BYTES:],
-            byteorder="little",
-        )
-        if actual_checksum != expected_checksum:
+        checksum_is_valid = validate_checksum(comm_from_pc)
```

Comment:
not for future: consider the more pythonic EAFP pattern of raising (and then handling) an error in the rare event that the checksum doesn't match, rather than the Java-esque LBYL pattern in current use